lte/gateway/c/core/oai/test/amf/test_amf_procedures.cpp

```diff
@@ -339,4 +357,345 @@ TEST_F(AMFAppProcedureTest, TestPDUSessionSetup) {
+  EXPECT_TRUE(rc == RETURNok);
+
+  ue_context_p = amf_ue_context_exists_amf_ue_ngap_id(ue_id);
+  // ue context should be exist
+  ASSERT_NE(nullptr, ue_context_p);
+  // smf context should be present
+  EXPECT_EQ(1, ue_context_p->amf_context.smf_ctxt_map.size());
+
+  /* Send duplicate uplink nas message for pdu session establishment request
+   * from UE */
+  rc = send_uplink_nas_pdu_session_establishment_request(
+      amf_app_desc_p, ue_id, plmn, ue_pdu_session_est_req_hexbuf,
+      sizeof(ue_pdu_session_est_req_hexbuf));
+  EXPECT_GE(rc, RETURNok);
+
+  // smf context should be present
+  EXPECT_EQ(1, ue_context_p->amf_context.smf_ctxt_map.size());
+  std::shared_ptr<smf_context_t> smf_ctx =
+      amf_get_smf_context_by_pdu_session_id(ue_context_p, 1);
+  EXPECT_EQ(1, smf_ctx->duplicate_pdu_session_est_req_count);
+
+  /* Send duplicate uplink nas message for pdu session establishment request
+   * from UE */
+  rc = send_uplink_nas_pdu_session_establishment_request(
+      amf_app_desc_p, ue_id, plmn, ue_pdu_session_est_req_hexbuf,
+      sizeof(ue_pdu_session_est_req_hexbuf));
+  EXPECT_GE(rc, RETURNok);
+
+  // smf context should be present
+  EXPECT_EQ(1, ue_context_p->amf_context.smf_ctxt_map.size());
+  smf_ctx = amf_get_smf_context_by_pdu_session_id(ue_context_p, 1);
+  EXPECT_EQ(2, smf_ctx->duplicate_pdu_session_est_req_count);
+
+  /* Send duplicate uplink nas message for pdu session establishment request
+   * from UE */
+  rc = send_uplink_nas_pdu_session_establishment_request(
+      amf_app_desc_p, ue_id, plmn, ue_pdu_session_est_req_hexbuf,
+      sizeof(ue_pdu_session_est_req_hexbuf));
+  EXPECT_GE(rc, RETURNok);
+
+  // smf context should be present
+  EXPECT_EQ(1, ue_context_p->amf_context.smf_ctxt_map.size());
+  smf_ctx = amf_get_smf_context_by_pdu_session_id(ue_context_p, 1);
+  EXPECT_EQ(3, smf_ctx->duplicate_pdu_session_est_req_count);
+
+  /* Send duplicate uplink nas message for pdu session establishment request
+   * from UE */
+  rc = send_uplink_nas_pdu_session_establishment_request(
+      amf_app_desc_p, ue_id, plmn, ue_pdu_session_est_req_hexbuf,
+      sizeof(ue_pdu_session_est_req_hexbuf));
+  EXPECT_GE(rc, RETURNok);
+
+  // smf context should be present
+  EXPECT_EQ(1, ue_context_p->amf_context.smf_ctxt_map.size());
+  smf_ctx = amf_get_smf_context_by_pdu_session_id(ue_context_p, 1);
+  EXPECT_EQ(4, smf_ctx->duplicate_pdu_session_est_req_count);
+
+  /* Send duplicate uplink nas message for pdu session establishment request
+   * from UE */
+  rc = send_uplink_nas_pdu_session_establishment_request(
+      amf_app_desc_p, ue_id, plmn, ue_pdu_session_est_req_hexbuf,
+      sizeof(ue_pdu_session_est_req_hexbuf));
+  EXPECT_GE(rc, RETURNok);
+
+  // smf context should be present
+  EXPECT_EQ(1, ue_context_p->amf_context.smf_ctxt_map.size());
+  smf_ctx = amf_get_smf_context_by_pdu_session_id(ue_context_p, 1);
+  EXPECT_EQ(4, smf_ctx->duplicate_pdu_session_est_req_count);
```

Comment:
LInes 630-682 can be replaced with a for loop

Reply:
```suggestion
  EXPECT_EQ(5, smf_ctx->duplicate_pdu_session_est_req_count);
```

Reply:
both review comments addressed and updated code